src/acpi.rs

```diff
@@ -419,3 +424,357 @@ pub fn process_msct() -> (
+    const ACPI_NFIT_MEM_FLUSH_FAILED: u32 = 0x1 << 2; /* 02: Platform flush failed */
+    const ACPI_NFIT_MEM_NOT_ARMED: u32 = 0x1 << 3; /* 03: Memory Device is not armed */
+    const ACPI_NFIT_MEM_HEALTH_OBSERVED: u32 = 0x1 << 4; /* 04: Memory Device observed SMART/health events */
+    const ACPI_NFIT_MEM_HEALTH_ENABLED: u32 = 0x1 << 5; /* 05: SMART/health events enabled */
+    const ACPI_NFIT_MEM_MAP_FAILED: u32 = 0x1 << 6; /* 06: Mapping to SPA failed */
+
+    unsafe {
+        assert_eq!((*entry).Header.Type, 1);
+        debug!(
+            "ACPI_NFIT_TYPE_MEMORY_MAP - Type {}, Length {}
+                NfitDeviceHandle: 0x{:x},
+                NfitDeviceHandle.DimmNumber: 0x{:x},
+                NfitDeviceHandle.MemChannel: 0x{:x},
+                NfitDeviceHandle.MemControllerId: 0x{:x},
+                NfitDeviceHandle.SocketId: 0x{:x},
+                NfitDeviceHandle.NodeControllerId: 0x{:x},
+                PhysicalId: {},
+                RegionId: {},
+                RangeIndex: {},
+                RegionIndex: {},
+                RegionSize: {},
+                RegionOffset: {},
+                Address: {},
+                InterleaveIndex: {},
+                InterleaveWays: {},
+                Flags: {}",
+            (*entry).Header.Type,
+            (*entry).Header.Length,
+            (*entry).DeviceHandle,
+            (*entry).DeviceHandle & ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK,
+            ((*entry).DeviceHandle >> 4) & ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK,
+            ((*entry).DeviceHandle >> 8) & ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK,
+            ((*entry).DeviceHandle >> 12) & ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK,
+            ((*entry).DeviceHandle >> 16) & ACPI_NFIT_DEVICE_HANDLE_12BIT_MASK,
+            (*entry).PhysicalId,
+            (*entry).RegionId,
+            (*entry).RangeIndex,
+            (*entry).RegionIndex,
+            (*entry).RegionSize,
+            (*entry).RegionOffset,
+            (*entry).Address,
+            (*entry).InterleaveIndex,
+            (*entry).InterleaveWays,
+            (*entry).Flags,
+        );
+    }
+}
+
+// NFIT subtable type = 0x2
+fn parse_nfit_interleave_structure(entry: *const ACPI_NFIT_INTERLEAVE) {
+    unsafe {
+        assert_eq!((*entry).Header.Type, 2);
+        debug!(
+            "ACPI_NFIT_TYPE_INTERLEAVE - Type {}, Length {}
+            InterleaveStructureIndex: {:#x}
+            NumberOfLinesDescribed: {:#x}
+            LineOffset: {:#x}
+            LineSize: {:#x}",
+            (*entry).Header.Type,
+            (*entry).Header.Length,
+            (*entry).InterleaveIndex,
+            (*entry).LineCount,
+            (*entry).LineOffset[0],
+            (*entry).LineSize
+        );
+    }
+}
+
+// NFIT subtable type = 0x4
+fn parse_nfit_control_region_structure(entry: *const ACPI_NFIT_CONTROL_REGION) {
+    unsafe {
+        assert_eq!((*entry).Header.Type, 4);
+        debug!(
```

Comment:
Can we just implement debug on type to put actual print logic there?

Reply:
Should be done in the acpica-sys repo?